spec/acceptance/suites/dnfmodule/redis_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'spec_helper_acceptance'
+
+describe 'redis', if: %w[centos redhat].include?(os[:family]) && os[:release].to_i == 8 do
+  redis_name = 'redis'
+
+  it 'runs successfully' do
+    pp = <<-PUPPET
+      class { 'redis':
```

Comment:
I suspect it runs after a test which has already installed Redis and then the ensure doesn't match. I suspect you can use `package_ensure` to upgrade.
```suggestion
      class { 'redis':
        package_ensure => 'latest',
```